File: apps/chat/libs/chat-domain/src/main/java/com/example/chat/domain/user/UserId.java
package com.example.chat.domain.user;

import java.util.UUID;

import org.springframework.lang.NonNull;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.ToString;

/**
 * 사용자 ID (Value Object)
 */
@Getter
@EqualsAndHashCode
@ToString
public class UserId {
    @NonNull
    private final String value;

    private UserId(String value) {
        if (value == null || value.isBlank()) {
            throw new IllegalArgumentException("UserId cannot be null or blank");
        }
        this.value = value;
    }

    public static UserId of(String value) {
        return new UserId(value);
    }

    public static UserId generate() {
        return new UserId(UUID.randomUUID().toString());
    }
}

File: apps/chat/libs/chat-domain/src/main/kotlin/com/example/chat/domain/user/UserId.kt
package com.example.chat.domain.user

import java.util.UUID

/**
 * 사용자 ID (Value Object)
 */
data class UserId(
    val value: String
) {
    init {
        require(value.isNotBlank()) { "UserId cannot be null or blank" }
    }

    companion object {
        @JvmStatic
        fun of(value: String): UserId = UserId(value)

        @JvmStatic
        fun generate(): UserId = UserId(UUID.randomUUID().toString())
    }
}
